### PR TITLE
[typescript-migration] initial ts config and the first calendar_container.jsx module

### PR DESCRIPTION
**Pull request by mirus-ua**
---
name:  initial ts config and the first calendar_container.jsx module
---

## Description
**Linked issue**: https://github.com/Hacker0x01/react-datepicker/issues/4700
**Problem**

**Changes**
Was added initial ts config and rollup plugins + migrated the first functional component to TS

**To reviewers:**
I've added babel preset-typescript instead of using ts-jest because right now, during the migration, better to only transpile code and not check types validity in tests 

## Screenshots

Знімок екрана 2024-04-18 о 16 36 00
Знімок екрана 2024-04-18 о 16 36 19

## Contribution checklist
- [x] I have followed the [contributing guidelines](https://github.com/Hacker0x01/react-datepicker/blob/main/CONTRIBUTING.md).
- [x] I have added sufficient test coverage for my changes.
- [x] I have formatted my code with Prettier and checked for linting issues with ESLint for code readability.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
+    "allowSyntheticDefaultImports": true,
+    "noFallthroughCasesInSwitch": true,
+    "noUnusedParameters": true,
+    "noUnusedLocals": true,
+    "declarationDir": "dist/types",
+    "declaration": true,
+    "strict": true,
```
[the remark this answers is not in the document]
Comment (mirus-ua) [reply tsconfig.json:9]: added https://github.com/Hacker0x01/react-datepicker/pull/4704/commits/ac85514b703b6fc071b4658f70ab7bd55cfe87e7

File: package.json [hunk 2]
```diff
@@ -98,12 +102,15 @@
...
+    "build": "NODE_ENV=production yarn run build:js && NODE_ENV=production yarn run css:prod && NODE_ENV=production yarn run css:modules:dev && NODE_ENV=production yarn run css:dev && NODE_ENV=production yarn run css:modules:dev && NODE_ENV=production yarn run build:types",
+    "build-dev": "NODE_ENV=development yarn run js:dev && NODE_ENV=development yarn run css:dev && NODE_ENV=development yarn run css:modules:dev && NODE_ENV=production yarn run build:types",
     "css:prod": "sass --style compressed src/stylesheets/datepicker.scss > dist/react-datepicker.min.css",
     "css:modules:prod": "sass --style compressed src/stylesheets/datepicker-cssmodules.scss | tee dist/react-datepicker-cssmodules.min.css dist/react-datepicker-min.module.css",
     "css:dev": "sass --style expanded src/stylesheets/datepicker.scss > dist/react-datepicker.css",
     "css:modules:dev": "sass --style expanded src/stylesheets/datepicker-cssmodules.scss | tee dist/react-datepicker-cssmodules.css dist/react-datepicker.module.css",
+    "type-check": "tsc --noEmit",
+    "type-check:watch": "npm run type-check -- --watch",
+    "build:types": "tsc --emitDeclarationOnly",
     "build:js": "rollup -c",
```
[the remark this answers is not in the document]
Comment (mirus-ua) [reply package.json:114]: What is your suggestion? `build:source` WDYT?

[the remark this answers is not in the document]
Comment (mirus-ua) [reply package.json]: here https://github.com/Hacker0x01/react-datepicker/pull/4704/commits/7a2fd9329d7fce33d51b328be9d8df73c1288566

**Comment by martijnrusschen** [comment]
Nice one! Glad to see this coming along!

**Pull request merged**